ompi/tools/mpirun/main.c

```diff
@@ -91,8 +91,9 @@ int main(int argc, char *argv[])
         fprintf(stderr, "prterun executable could not be found - unable to run\n");
         exit(1);
     }
-
+    pfx = opal_argv_join(pargs, ',');
     execve(truepath, pargs, environ);
+    fprintf(stderr, "The mpirun cmd failed to exec its actual executable - your application will NOT execute. Error: %s\n", strerror(errno));
```

Comment:
It might also be good to return a non-zero exit status here to indicate that `mpirun` failed.

Reply:
Sure - why not? 😄